### ISSUE TITLE
Deprecate `login.nu` in favor of `$nu.is-login`?

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#14329]
### Related problem

The way that I understand it, users currently have two different ways that they can specify commands to run at startup of a *login shell*:

1. Place the command(s) in `login.nu` which is automatically run after `config.nu` for login shells
2. Wrap commands in `if $nu.is-login { <commands> }` in the `config.nu`

The second form shares the same style as `$nu.is-interactive`.

### Describe the solution you'd like

Can/should we deprecate `login.nu` in favor of conditional `$nu.is-login`?

This will simply the startup code somewhat if we can do it.

**Comment by fdncred** [comment]
if we had an `if --env` I'd support this but since we don't yet, it makes it harder to do anything with `if` that affects the environment. It seems like we'd run into the same old problem of, "ya, this kind of works but it's scoped so it doesn't really do all that i need it to" which is what we hear pretty frequently.

**Comment by NotTheDr01ds** [comment]
Ah, Thanks - That makes perfect sense.  I clearly forgot about the scoping problem.  I'll hold this open with the understanding that we need `if --env` before we could possibly deprecate it.

**Comment by IanManske** [comment]
Regular `if` should work, no? Env is only scoped for closures and custom commands. (Another reason to differentiate blocks and closures via syntax.)

**Comment by NotTheDr01ds** [comment]
@IanManske I'm sure you're right.  I'll test it to be certain, but I'm sure you are right :-)

**Comment by fdncred** [comment]
@IanManske Is that fixed? People have wanted to do this (below) for the longest time and it never worked.

```nushell
if $nu.os-info.name == windows {
  use some_windows_thing.nu
  source windows.nu
} else if $nu.os-info.name == linux {
  use some_linux_thing.nu
  source linux.nu
} else {
  use some_mac_thing.nu
  source mac.nu
}
```

**Comment by IanManske** [comment]
Oh imports are a separate thing, I'm not sure what the current state/behavior is.

**Comment by NotTheDr01ds** [comment]
What we kind of need there is (hand-waving), `export <def>/<alias>/<etc>` to promote outside the current block no matter where it is called from.  For example:

```nushell
if $nu.os-info.name == windows {
  export use some_windows_thing.nu
}
```

**Comment by Bahex** [comment]
This is actually possible, just not immediately obvious.

```nushell
const file = if $nu.os-info.name == windows {
  "windows.nu"
} else if $nu.os-info.name == linux {
  "linux.nu"
} else {
  "mac.nu"
}

source $file
```

But an extra (empty) file is needed.
```nushell
source (if $nu.is-login { "foo.nu" } else { "empty.nu" })
```

**Comment by fdncred** [comment]
agreed. however, imo we need a `source --env`/`use --env` and `if --env` as a way to escape the scope.
